mlflow/genai/scorers/builtin_scorers.py

```diff
@@ -366,38 +381,106 @@ def guideline_adherence(
-
 
-class _RelevanceToQuery(_BaseBuiltInScorer):
-    name: str = "relevance_to_query"
+    name: str = "guideline_adherence"
+    global_guidelines: Optional[list[str]] = None
     required_columns: set[str] = {"inputs", "outputs"}
 
-    def __call__(self, *, inputs: Any, outputs: Any) -> Assessment:
-        """Evaluate relevance to the user's query."""
-        return super().__call__(inputs=inputs, outputs=outputs)
+    def update_evaluation_config(self, evaluation_config) -> dict:
+        # Metric name should always be "guideline_adherence" regardless of the custom name
+        config = deepcopy(evaluation_config)
+        metrics = config.setdefault(GENAI_CONFIG_NAME, {}).setdefault("metrics", [])
+        if "guideline_adherence" not in metrics:
+            metrics.append("guideline_adherence")
+
+        # If global guidelines are specified, add it to the config
+        if self.global_guidelines:
+            # NB: The agent eval harness will take multiple global guidelines in a dictionary format
+            #   where the key is the name of the global guideline judge. Therefore, when multiple
+            #   judges are specified, we merge them into a single dictionary.
+            #   https://docs.databricks.com/aws/en/generative-ai/agent-evaluation/llm-judge-reference#examples
+            global_guidelines = config[GENAI_CONFIG_NAME].get("global_guidelines", {})
+            global_guidelines[self.name] = self.global_guidelines
+            config[GENAI_CONFIG_NAME]["global_guidelines"] = global_guidelines
+
+        return config
+
+    def validate_columns(self, columns: set[str]) -> None:
+        super().validate_columns(columns)
+        # If no global guidelines are specified, the guidelines must exist in the input dataset
+        if not self.global_guidelines and "expectations/guidelines" not in columns:
+            raise MissingColumnsException(self.name, ["expectations/guidelines"])
+
+    def __call__(
+        self,
+        *,
+        inputs: Any,
+        outputs: Any,
+        guidelines: dict[str, list[str]],
+        guidelines_context: dict[str, Any],
+    ) -> Assessment:
+        """Evaluate adherence to specified guidelines."""
+        return super().__call__(
+            inputs=inputs,
+            outputs=outputs,
+            guidelines=guidelines,
+            guidelines_context=guidelines_context,
+        )
+
+    def with_config(
+        self,
+        *,
+        name: str = "guideline_adherence",
+        global_guidelines: Optional[list[str]] = None,
+    ) -> "GuidelineAdherence":
+        """
+        Get a new scorer instance with the given name and global guidelines.
+
+        Args:
+            name: The name of the scorer. Default is "guideline_adherence".
+            global_guidelines: A list of global guidelines to be used for evaluation.
+                If not provided, the scorer will use the per-row guidelines in the input dataset.
```

Comment:
Let's make very certain that the docs that accompany these API changes have examples of embedding the columnar approach to guidelines (likely less common) and the global guidelines exposed via this API to show both happy paths for eval usage.